processor/attributesprocessor/attributes.go

```diff
@@ -42,6 +44,12 @@ type attributesConfig struct {
 type attributeAction struct {
 	Key           string
 	FromAttribute string
+	// Compiled regex if provided
+	Regex *regexp.Regexp
+	// Attribute names extracted from the regexp's subexpressions.
+	AttrNames []string
+	// Number of non empty strings in above array
+
```

Comment:
I think we are overloading too much this struct. It feel to me that we have 3 different types of operations:
* SetValue -> {AttributeValue, Action}
* SetValueFromAttribute -> {FromAttribute, Action}
* SetValuesFromRegexp -> ....

I think it is time to split this `attributeAction` into multiple types of `Operations`

Reply:
I agree it's overloaded and might make sense to split. In that case though, what do you think of also splitting up the **delete** and **hash** operations to keep things consistent?

Reply:
Since this could result in a breaking change of the config, issue #1170 has been opened to address these type of questions/statements